engine/apps/schedules/tasks/refresh_ical_files.py

```diff
@@ -84,6 +88,8 @@ def refresh_ical_file(schedule_pk):
     # update cached schedule on-call users
     update_cached_oncall_users_for_schedule(schedule)
 
+    check_gaps_and_empty_shifts_in_schedule.apply_async((schedule_pk,))
+
     if run_task:
         notify_about_empty_shifts_in_schedule_task.apply_async((schedule_pk,))
```

Comment:
These tasks are not about updating these fields, they notify users in Slack about gaps and empty shifts if ical was changed

Reply:
Hm.. but these tasks, when run, will re-check for gaps and empty shifts (and update the fields too, it seems), so in some cases we will be doing the work twice?

Reply:
yeah... 🤔 but I don't see how to avoid this without more detailed refactoring, since these `notfify` tasks use different types of events with different fields: `ScheduleEvent` and `EmptyShifts`. For regular update I use `ScheduleEvent` for both checks and parse calendars only once because we don't need an additional information about empty events there.

Reply:
Ok, makes sense. I would at least leave some comment/TODO/issue that we would like to refactor this at some point.